src/main/java/me/contaria/seedqueue/gui/wall/SeedQueuePreview.java

```diff
@@ -202,6 +202,10 @@ public boolean isCooldownReady() {
         return System.currentTimeMillis() - this.cooldownStart >= SeedQueue.config.resetCooldown;
     }
 
+    public void resetCooldown() {
+        this.cooldownStart = System.currentTimeMillis();
```

Comment:
Cooldown starts are meant to be populated in `SeedQueueWallScreen#populateResetCooldowns` because that method gets called after the new frame is actually rendered onto the screen instead of doing it during rendering. To reset the cooldown just set it to `Long.MAX_VALUE`, that will cause its cooldown to be repopulated at the end of the frame.

Reply:
> Cooldown starts are meant to be populated in `SeedQueueWallScreen#populateResetCooldowns` because that method gets called after the new frame is actually rendered onto the screen instead of doing it during rendering. To reset the cooldown just set it to `Long.MAX_VALUE`, that will cause its cooldown to be repopulated at the end of the frame.

interesting (implemented changes)